### ISSUE TITLE
ZEN inline 8.0.1 and Twilio captiveportal exception "No authentication context"

### ISSUE THREAD

**Issue by f1iqf** [inverse-inc/packetfence#3250]
The problem occurs when using PacketFence ZEN 8.0.1 in inline configuration with the Twilio authentication at the captiveportal. If using simple SMS authentication, no problem. However, if ticking the box "Create local account" to save the email of the user, the captive portal fail with the error :

Caught exception in captiveportal::Controler::(...)>dynamic_application "No authentication context provided at /usr/local/pf/lib/pf/authentication.pm line 299."

The problem can be easyly reproduced with any new ZEN installation and a Twilio free test account.

Issue created following a question on packetfence-users@lists.sourceforge.net and on advice from M. Fabrice Durand.

![zeninlinetwilio](https://user-images.githubusercontent.com/39140742/40443472-70ad02da-5e94-11e8-945b-49571bcfd32b.PNG)

**Comment by f1iqf** [comment]
Commenting out lines 299 and 301 in /local/usr/pf/lib/pf/authentication.pm seems to "fix" the problem, however this is kind of a brutal fix. I don't seem to be able to find any adverse effect in the fix, which of course does not mean there is none ;-)
- The captive portal alllows connexion
- The local user is updated with correct phone number and email address

Untill there is a more complete fix, this allows me to pursue my proof of concept project :)

**Comment by julsemaan** [comment]
Can you attempt to apply the following fix and let me know if it works:
```
cd /usr/local/pf/
curl https://github.com/inverse-inc/packetfence/commit/0385f54a0fb47713da852fc15d9e15f187e0a411.diff | patch -p1
bin/pfcmd service httpd.portal restart
```

I can then put this in the maintenance

**Comment by f1iqf** [comment]
That works, Thank You !

**Comment by f1iqf** [comment]
Hi Julien,


Your solution works perfectly - thank you very much !

Should I close the github issue ?


Best regards,

Nicolas


De : Julien Semaan <notifications@github.com>
Envoyé : 23 mai 2018 16:23
À : inverse-inc/packetfence
Cc : f1iqf; Author
Objet : Re: [inverse-inc/packetfence] ZEN inline 8.0.1 and Twilio captiveportal exception "No authentication context" (#3250)


Can you attempt to apply the following fix and let me know if it works:

cd /usr/local/pf/
curl https://github.com/inverse-inc/packetfence/commit/0385f54a0fb47713da852fc15d9e15f187e0a411.diff | patch -p1
bin/pfcmd service httpd.portal restart


I can then put this in the maintenance

**Comment by julsemaan** [comment]
I'll cherry-pick the fix into maintenance/8.0 so its available through the maintenance and close this today

**Comment by julsemaan** [comment]
Pushed in maintenance/8.0

Closing
